### ISSUE TITLE
Cmd+Shift+Enter keyboard shortcut does not work in code cells

### ISSUE THREAD

**Issue by seeM** [posit-dev/positron#3509]
## Positron Version:
Positron Version: 2024.06.0 (Universal) build 40

## Steps to reproduce the issue:

1. Create a file with a code cell:
    ```python
    # %%
    ```
2. Press Cmd+Shift+Enter. The code cell will be executed (the Shift+Enter shortcut swallows the Cmd+Shift+Enter one).

## What did you expect to happen?
The file should have been executed i.e. you should see `%run ...` in the console.

## Were there any error messages in the output or Developer Tools console?
No.

**Comment by juliasilge** [comment]
Related to #8637

**Comment by juliasilge** [comment]
I would advocate that we have the right behavior here, and should not make a change.

- If you are in a code cell, either in `.R` or `.py`, then <kbd>Cmd/Ctrl+Shift+Enter</kbd> runs your current code cell via `positron.runCurrentCell`. This has a nice parallel in `.qmd` with `quarto.runCurrentCell` which has the same shortcut. We did work in #8888 to make sure that this works this way.
- If you are _not_ in a code cell but in a `.R.` or `.py`file, then then <kbd>Cmd/Ctrl+Shift+Enter</kbd> runs your current _file_ via `%run` or `source()`. I think it makes sense for `positron.runCurrentCell` to win over this one, if you have code cells.
- In either situation, <kbd>Cmd/Ctrl+Enter</kbd> runs the current _statement_.

Can I convince you on this, @seeM?

**Comment by seeM** [comment]
@juliasilge Ah ty, I hadn't considered Quarto consistency.

I agree but nervous that IIUC there is no keyboard shortcut to run/source a file in a `.R` or `.py` with code cells, which is a common enough action that we have an action bar button.

You can use <kbd>Opt+Cmd+R</kbd> to do `positron.runAllCells` but it doesn't match the behavior of `python.execInConsole`.

We could hijack <kbd>Opt+Cmd+R</kbd> to do `python.execInConsole`/`r.sourceCurrentFileWithEcho` in `.py`/`.R` with code cells to keep consistent with Quarto? Or is there another Quarto action that is a closer parallel e.g. render or preview?

**Comment by juliasilge** [comment]
Quarto does provide `quarto.preview` via <kbd>Cmd+Shift+K</kbd>, but it _also_ provides `quarto.previewScript` at the same keyboard shortcut which is [supposed to apply to these kinds of documents with code cells](https://github.com/quarto-dev/quarto/blob/591b3520eafbb4da7b26b9f31aac6948801f19d8/apps/vscode/src/core/doc.ts#L126C10-L126C32) so that's not really an option for us to use.

I don't think we should prioritize `python.execInConsole` over `positron.runAllCells` in a file with code cells. I tend to think the accessibility on the editor action bar and command palette is enough combined with the ability for folks to customize their keyboard shortcuts, if they really want the `%run` behavior for a file with code cells via a keyboard shortcut. What do you think?

**Comment by seeM** [comment]
Yep, seems reasonable to me! We can always revisit based on user feedback

**Comment by juliasilge** [comment]
OK, we'll look for feedback from folks about their preferences when it comes to keyboard shortcuts in files with code cells!
